src/ripple/rpc/handlers/ValidatorInfo.cpp

```diff
@@ -0,0 +1,61 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Dev Null Productions
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/misc/ValidatorKeys.h>
+#include <ripple/basics/base64.h>
+
+namespace ripple {
+Json::Value doValidatorInfo (RPC::Context& context)
+{
+    // return error if not configured as validator
+    if(context.app.getValidationPublicKey().empty())
+      return RPC::not_validator_error ();
+
+    Json::Value ret;
+
+    auto pk = context.app.getValidationPublicKey();
+
+    // assume pk is ephemeral key, get master key
+    auto mk = context.app.validatorManifests().getMasterKey(pk);
+    ret[jss::master_key] = toBase58(TokenType::NodePublic, mk);
+
+    // pk is maskter key, eg no ephemeral key, eg no manifest, just return
+    if(mk == pk)
+      return ret;
+
+    // lookup ephemeral key
+    auto ek = context.app.validatorManifests().getSigningKey(mk);
+    ret[jss::ephemeral_key] = toBase58(TokenType::NodePublic, ek);
+
+    auto manifest = context.app.validatorManifests().getManifest(mk);
+    if(manifest)
+      ret[jss::manifest] = base64_encode((std::uint8_t const*)manifest->c_str(),
+                                                             manifest->length());
+
+    auto seq = context.app.validatorManifests().getSequence(mk);
+    if(seq)
```

Comment:
`seq` scope could be narrowed to the if statement

Also one space after `if` and condition and indent four spaces